mission_control/navigator_missions/nav_missions/symbol_mission.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+import txros
+import rospy
+import std_srvs.srv
+from navigator_msgs.msg import DockShape
+import numpy
+from geometry_msgs.msg import Point
+import sensor_msgs.msg
+import nav_msgs.msg
+
+import cv2
+
+from twisted.internet import defer
+
+
+
+error_threshold = 0.01
+move_per_iteration = 1.0;
+#~ square = numpy.array([Point(-1,-1,0), Point(-1,1,0), Point(1,1,0), Point(1,-1,0)])
+#~ square = numpy.array([-1,-1,0,-1,1,0,1,1,0,1,-1,0])
+square = numpy.array([[0, 0, 0],[ 1, 0, 0], [0, 1, 0],[ 1, 1, 0]]).astype(numpy.float64)
+#~ K = numpy.array(numpy.mat("1 0 0;0 1 0;0 0 1"))
+K = numpy.array([[1,0,0],[0,1,0],[0,0,1]]).astype(numpy.float64)
+
+
+def boundingRect(points):
+    maxX = 0
+    minX = 2000
+    maxY = 0
+    minY = 2000
+    for i in range(len(points)):
+        if(points[i].x > maxX):
+            maxX = points[i].x;
+        if(points[i].y > maxY):
+            maxY = points[i].y;
+        if(points[i].x < minX):
+            minX = points[i].x;
+        if(points[i].y < minY):
+            minY = points[i].y;
+    return numpy.array([maxX, maxY, minX, minY, minX, maxY, maxX, minY])
+    #~ return [Point(minX, maxY, 0), Point(maxX, minY,0), Point(maxX, maxY,0), Point(minX, minY,0)]
+
+def callback(data):
+    #~ K = numpy.reshape(data.K, (3,3))
+    print K
+    
+@txros.util.cancellableInlineCallbacks
+def listener(navigator):
+    camerainfoSub = yield navigator.nh.subscribe("/side_camera/camera_info", sensor_msgs.msg.CameraInfo)
+    caminfo = yield b.get_next_message()
+    #~ K = caminfo.K
+    #print caminfo.K
+    defer.returnValue(caminfo.K)
+    #~ rospy.spin()
+
+@txros.util.cancellableInlineCallbacks
+def main(navigator):
+    
+    #~ shooterFire = rospy.ServiceProxy("/shooter/fire", std_srvs.srv.Trigger)
+    #~ 
+    #~ cameraInfo = rospy.ServiceProxy("/side_camera/camera_info", sensor_msgs.msg.CameraInfo)
+    #~ K = yield listener(navigator)
+    #~ yield navigator.nh.sleep(1)
+    position, orientiation = navigator.pose
+    
+    print orientiation
+    
+    print K
+    try:    
+        resp = yield navigator.vision_request("get_shape")
+        error = float(resp.symbol.CenterX)/resp.symbol.img_width - 0.5
```

Comment:
If you want to avoid having to cast one of these as floats, you can `from __future__ import division` at the top and then it'll automattically do float division rather than integer division.